Source/SteveSloth/Private/MyPermanentHealthPickup.cpp

```diff
@@ -0,0 +1,31 @@
+/****************************************************************************************
+ * Copyright:
+ * Name: Guy Lapensee
+ * Script: AMyPermanentHealthPickup
+ * Date: May 16, 2024
+ * Description:Leaf  pickup for 25 permanent health boost
+ * TODO: Permanently add health to the players health bar
+ * Known Bugs:
+ ****************************************************************************************/
+
+#include "MyPermanentHealthPickup.h"
+
+AMyPermanentHealthPickup::AMyPermanentHealthPickup()
+{
+	ItemClass = CreateDefaultSubobject<AItemBaseClass>(TEXT("Item Class"));
+	PlayerClass = CreateDefaultSubobject<AMyPlayer>(TEXT("Player Class"));
+
+	ItemName = "Leaf";
+	HealthIncrease = 25;
+	isItemUnlocked = true;
+
+	OnPickUp();
```

Comment:
Place this is the BeginPlay()

Reply:
not even in the begin play, it should be on collision with the player (dont need to worry about that yet, we still need to set up collisions properly)

Reply:
lol I though it was going to be an actual pickup like click on it.. forgot we are just going to run into it.. in that case yes should be in an on collision event